### PR TITLE
integration/qat: add test to check QAT

### LINKED ISSUE #3939: ci: add test to test QAT devices
cc @GabyCT

### PR DESCRIPTION
**Pull request by devimc**
Add integration test to test Intel® QAT devices

fixes #3939

Signed-off-by: Julio Montes <julio.montes@intel.com>

cc @GabyCT

### CODE REVIEW

File: integration/qat/qat_test.sh [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+	echo "Skip pmem test: ${ID} distro is not supported"
+	exit 0
+fi
+
+init() {
+	local qat_devices=$(for i in ${QAT_DEVICE_IDS[*]}; do lspci -d ${QAT_VENDOR_ID}:$i; done)
+	[ -n "${qat_devices}" ] || die "This system doesn't have any QAT device"
+}
+
+build_install_qat_kernel() {
```
Comment (fidencio) [inline integration/qat/qat_test.sh:49]: @devimc, @wainersm, can we try to cache this one as we already do with the "non-qat" kernel? :-)

Comment (GabyCT) [reply integration/qat/qat_test.sh:49]: @fidencio I am not sure that is possible, I believe you need special hardware for that

**Review by GabyCT — APPROVED** [review]

**Comment by GabyCT** [comment]
@devimc I think what is missing is to add this to `run.sh`, if you want this can be submitted as it is and then I can add the test

File: integration/qat/qat_test.sh (hunk 1 above)
Comment (devimc) [reply integration/qat/qat_test.sh:49]: it's possible but we'll need kernel sources to build the QAT modules and place it into the rootfs image, so kernel + kernel sources maybe too much to cache :man_shrugging:

Comment (fidencio) [reply integration/qat/qat_test.sh:49]: Ack.

Out of curiosity, are we adding a machine that supports QAT to our pool so we run those tests regularly?

Comment (devimc) [reply integration/qat/qat_test.sh:49]: For now we'll testing this internally...

File: integration/qat/qat_test.sh [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+# List of QAT device IDs
+QAT_VENDOR_AND_ID_VF=
+QAT_DEVICE_ID=37c8
+QAT_VENDOR_ID=8086
+QAT_DEV_ADDR=
+SSL_IMAGE_TAG=openssl-qat-engine
+CTR_RUNTIME="${CTR_RUNTIME:-io.containerd.run.kata.v2}"
+
+if [ "$arch" == "aarch64" ]; then
+	echo "Skip pmem test: $arch can't ensure data persistence for the lack of libpmem support"
```
Comment (chavafg) [inline integration/qat/qat_test.sh:33]: `s/pmem/QAT`?

File: integration/qat/qat_test.sh [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+	exit 0
+fi
+
+if [ "${ID}" != "ubuntu" ]; then
+	echo "Skip pmem test: ${ID} distro is not supported"
```
Comment (chavafg) [inline integration/qat/qat_test.sh:38]: same as above

**Comment by devimc** [comment]
@chavafg thanks, changes applied

/test

File: integration/qat/qat_test.sh [hunk 4]
```diff
@@ -0,0 +1,142 @@
+#!/bin/bash
+#
+# Copyright (c) 2021 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# WARNING: This script only support Intel® QAT c6xx devices
+#
+
+set -x
```
Comment (dborquez) [inline integration/qat/qat_test.sh:10]: just remove this

**Review by dborquez — APPROVED** [review]
LGTM, thanks @devimc

**Pull request merged**